src/entities/EditorDocument/EditorDocument.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { EditorDocument } from './index';
+import { BlockNode, BlockNodeName, createBlockNodeName } from '../BlockNode';
+
+const createBlock = ({ name, parent }: { name?: BlockNodeName, parent: EditorDocument }): BlockNode => {
+  return new BlockNode({
+    name: name || createBlockNodeName('block'),
+    parent,
+    children: {},
+  });
+};
+
+describe('EditorDocument', () => {
+  let document: EditorDocument;
+  let blocks: BlockNode[];
+
+  beforeEach(() => {
+    document = new EditorDocument({
+      children: [],
+      properties: {
+        readOnly: false,
+      },
+    });
+
+    const countOfBlocks = 3;
+
+    blocks = [];
+
+    for (let i = 0; i < countOfBlocks; i++) {
+      const block = createBlock({
+        parent: document,
+      });
+
+      document.addBlock(block);
+      blocks.push(block);
+    }
+  });
+
+  describe('length', () => {
+    it('should return the number of blocks in the document', () => {
+      // Arrange
+      const expected = 3;
+      const document1 = new EditorDocument({
+        children: [],
+        properties: {
+          readOnly: false,
+        },
+      });
+
+      for (let i = 0; i < expected; i++) {
+        const block = createBlock({
+          parent: document1,
+        });
+
+        document1.addBlock(block);
+      }
+
+      // Act
+      const actual = document1.length;
+
+      // Assert
+      expect(actual).toBe(expected);
+    });
+  });
+
+  describe('addBlock', () => {
+    it('should add the block to the end of the document if index is not provided', () => {
+      // Arrange
+      const block = createBlock({
+        parent: document,
+      });
+
+      // Act
+      document.addBlock(block);
+
+      // Assert
+      const lastBlock = document.getBlock(document.length - 1);
+
+      expect(lastBlock).toBe(block);
+    });
+
+    it('should add the block to the beginning of the document if index is 0', () => {
+      // Arrange
+      const block = createBlock({
+        parent: document,
+      });
+
+      // Act
+      document.addBlock(block, 0);
+
+      // Assert
+      expect(document.getBlock(0)).toBe(block);
+    });
+
+    it('should add the block to the specified index', () => {
+      // Arrange
+      const block = createBlock({
+        parent: document,
+      });
+
+      // Act
+      document.addBlock(block, 1);
+
+      // Assert
+      expect(document.getBlock(1)).toBe(block);
+    });
+
+    it('should add the block to the end of the document', () => {
```

Comment:
name is not clear. What's the difference between this and `should add the block to the end of the document if index is not provided`?